cirq/ops/thermal_relaxation_channel.py

```diff
@@ -0,0 +1,229 @@
+
+
+@value.value_equality
+class ThermalRelaxationChannel(gate_features.SingleQubitGate):
+    """Dampen qubits at varying rates through non ideal disipation.
+
+    This channel models the effect of energy dissipation into the environment
+    as well as the environment depositing energy into the system at varying
+    rates.
+    """
+
+    def __init__(self, p: float, gamma: float, beta: float) -> None:
+        r"""The thermal relaxation channel.
+
+        Construct a channel to model energy dissipation into the environment
+        as well as the environment depositing energy into the system. The
+        probabilities with which the energy exchange occur are given by `gamma`
+        and `beta` with the probability of the environment being not excited is
+        given by `p`.
+
+        The stationary state of this channel is the diagonal density matrix
+        with probability `p` of being |0⟩ and probability `1-p` of being |1⟩.
+
+        Recall the kraus opeartors for the `GeneralizedAmplitudeDampingChannel`:
+
+            $$
+            \begin{aligned}
+            M_0 =& \sqrt{p} \begin{bmatrix}
+                                1 & 0  \\
+                                0 & \sqrt{1 - \gamma}
+                            \end{bmatrix}
+            \\
+            M_1 =& \sqrt{p} \begin{bmatrix}
+                                0 & \sqrt{\gamma} \\
+                                0 & 0
+                           \end{bmatrix}
+            \\
+            M_2 =& \sqrt{1-p} \begin{bmatrix}
+                                \sqrt{1-\gamma} & 0 \\
+                                 0 & 1
+                              \end{bmatrix}
+            \\
+            M_3 =& \sqrt{1-p} \begin{bmatrix}
+                                 0 & 0 \\
+                                 \sqrt{\gamma} & 0
+                             \end{bmatrix}
+            \end{aligned}
+            $$
+
+        This has a choi representation of:
+
+            $$
+             \begin{aligned}
+                CHO =&  \begin{bmatrix}
+                            1 - (1-p) \gamma & 0 & 0 & \sqrt{1 - \gamma} \\
+                            0 & (1-p) \gamma 0 & 0 \\
+                            0 & 0 & p \gamma & 0 \\
+                            \sqrt{1 - \gamma} & 0 & 0 & 1 - p \gamma
+                        \end{bmatrix}
+             \end{aligned}
+            $$
+
+        Where each 2x2 block can be thought of as how the channel acts on each
+        basis element in the space of density matrices.
+
+        For the `ThermalRelaxationChannel` channel we simply substitute `beta`
+        in for `gamma` on the off diagonal elements of the above choi matrix
+        and then evolve the channing use the kraus operators found by sqrt
+        factorization of the choi matrix.
+
+        Args:
+            gamma: the probability of the interaction being dissipative.
+            beta: the probability of the interaction being dissipative.
```

Comment:
beta and gamma are exactly the same according to this doc

Reply:
fixed.